tests/kernel/smp/src/main.c

```diff
@@ -25,6 +25,12 @@
 #define THREAD_DELAY 1
 #define SLEEP_MS_LONG ((int)(15000 * RUN_FACTOR))
 
+/* We iterate priorities in spawn_threads() - so we need to have enough priorities */
+#define SPAWN_THREADS_MAX_PRIO 16
+BUILD_ASSERT(SPAWN_THREADS_MAX_PRIO >= CONFIG_MP_MAX_NUM_CPUS);
```

Comment:
This is a bit of a shortcut. Alternatively, make a Kconfig range option local to this test, where the lower bound is here, and the upper bound is one of the values below.
```suggestion
#define SPAWN_THREADS_MAX_PRIO CONFIG_MP_MAX_NUM_CPUS
```